packages/ngx-deploy-npm/src/generators/install/generator.ts

```diff
@@ -8,86 +8,47 @@ import type { Tree } from '@nx/devkit';
 
 import type { InstallGeneratorOptions } from './schema';
 import { DeployExecutorOptions } from '../../executors/deploy/schema';
-import {
-  allProjectsAreValid,
-  buildInvalidProjectsErrorMessage,
-  determineWhichProjectsAreInvalid,
-  isProjectAPublishableLib,
-  normalizeOptions,
-} from './utils';
+import { normalizeOptions } from './utils';
+import { isProjectAPublishableLib } from '../../utils';
 
 export default async function install(
   tree: Tree,
   rawOptions: InstallGeneratorOptions
 ) {
-  let libs = getBuildableLibraries(tree);
   const options = normalizeOptions(rawOptions);
 
-  // If there is no libraries to install throw an exception
-  if (libs.size === 0) {
-    throw new Error('There is no publishable libraries in this workspace');
-  }
-
-  if (options.projects && options.projects.length > 0) {
-    // if there is projects that doesn't exists, throw an error indicating which projects are invalid
-    if (!allProjectsAreValid(options.projects, libs)) {
-      const invalidProjects = determineWhichProjectsAreInvalid(
-        options.projects,
-        libs
-      );
-
-      throw new Error(buildInvalidProjectsErrorMessage(invalidProjects));
-    }
+  const selectedLib = getProjects(tree).get(options.project);
 
-    const selectedLibs = new Map<string, ProjectConfiguration>();
-    options.projects.forEach(project => {
-      const lib = libs.get(project);
-
-      if (lib) {
-        selectedLibs.set(project, lib);
-      }
-    });
-
-    libs = selectedLibs;
+  if (selectedLib === undefined) {
+    throw new Error(
+      `The project ${options.project} doesn't exist on your workspace`
+    );
   }
 
-  Array.from(libs.entries()).forEach(([libKey, libConfig]) => {
-    if (libConfig.targets) {
-      const executorOptions: DeployExecutorOptions = {
-        access: options.access,
-        ...setUpProductionModeIfHasIt(libConfig),
-      };
+  if ((await isProjectAPublishableLib(selectedLib)) === false) {
+    throw new Error(
+      `The project ${options.project} is not a publishable library`
+    );
+  }
 
-      libConfig.targets.deploy = {
-        executor: 'ngx-deploy-npm:deploy',
-        options: executorOptions,
-      };
+  const executorOptions: DeployExecutorOptions = {
+    distFolderPath: options.distFolderPath,
+    access: options.access,
+    ...setUpProductionModeIfHasIt(selectedLib),
+  };
+
+  if (selectedLib.targets) {
```

Comment:
If there is no targets, we should create the targets instead of avoiding creating a new one

Reply:
Done